airbyte-cdk/python/airbyte_cdk/connector_builder/models.py

```diff
@@ -39,6 +39,8 @@ class StreamReadSlices:
 class LogMessage:
     message: str
     level: str
+    internal_message: Optional[str] = None
```

Comment:
The trade off here is that since this does not map to the protocol error message model, when we move to using source-declarative-manifest container as manifest runner for Builder, we'd need to make sure that the protocol has these fields, otherwise error reporting will break. 

Should we make the work and align these models in this file with protocol-level models now instead? Or ship the small improvement and record a todo for later? 
/cc @bnchrch

Reply:
Thats a good callout Natik! 

Looking into it, this does already align with the protocol model  `AirbyteTraceMessage`

https://github.com/airbytehq/airbyte-protocol/blob/e24ee151574a7f0f4e997b90063d7c9ffed2b158/protocol-models/src/main/resources/airbyte_protocol/v0/airbyte_protocol.yaml#L256

❓ @lmossman can you confirm Im reading this as this is already a protocol blessed type?

If so should `LogMessage` become a union type of `AirbyteTraceMessage` and `AirbyteLogMessage`

Reply:
> The trade off here is that since this does not map to the protocol error message model, when we move to using source-declarative-manifest container as manifest runner for Builder, we'd need to make sure that the protocol has these fields, otherwise error reporting will break.

This `LogMessage` class is only used in the `message_grouper` logic which wraps a call to the `read` protocol method call.

Even if we swap out the runtime to use source-declarative-manifest to execute the read, we will still need all of this extra `message_grouper` logic around it to create the `StreamRead` output object that the connector builder server expects, and it shouldn't need to be changed because the `get_message_groups` function _does_ operate on the airbyte protocol already, and is just normalizing those airbyte protocol messages into a form that is easier understood by the connector builder server.

If we don't want to directly execute any python anymore at all and therefore delete this `message_grouper` logic, then the connector builder server will require a bunch of other changes anyway to do similar logic to what that python code is doing now.

So, I think I'd prefer to keep this PR simple right now, especially because I don't think we can easily reference the protocol AirbyteLogMessage and AirbyteTraceMessage in our connector-builder-server and airbyte-server openapi specs, and I'd rather not repeat those entire schemas as they are more complicated than the LogMessage schema I've defined here.

But let me know if anything I said above sounds off @natikgadzhi @bnchrch !